pkg/instance/instance.go

```diff
@@ -959,6 +977,37 @@ func (i *Instance) SetPrometheusRemoteWriteExporter(endpoint string) error {
 	return nil
 }
 
+// TsharkCollectorEnabled returns true if the tshark collector is enabled
+func (i *Instance) TsharkCollectorEnabled() bool {
+	return i.tsharkCollectorConfig != nil
+}
+
+// EnableTsharkCollector enables the tshark collector for the instance
+// This function can only be called in the state 'Preparing' or 'Committed'
+func (i *Instance) EnableTsharkCollector(volumeSize, s3AccessKey, s3SecretKey, s3Region, s3BucketName, s3KeyPrefix string) error {
```

Comment:
Would be better to define a config struct for this and pass it instead of too many arguments. That way we can also change the type, order and/or the number of arguments without braking the code in future,

Reply:
good follow up imo